# Fungal microbiomes: strategies for improving taxonomy resolution of environmental sequences

Table of Contents
--------
* [**Authors**](#authors)
* [**Abstract**](#abstract)
* [**Study Purpose**](#study-purpose)
* [**Pipeline**](#pipeline)
* [**Script Purpose**](#script-purpose)
* [**System Requirements**](#sys-req)


<a name="authors"></a>
Authors
--------
* [Kristi Gdanetz MacCready](https://github.com/gdanetzk)
* [Natalie Vande Pol](https://github.com/natalie-vandepol)
* [Gregory Bonito](https://www.researchgate.net/profile/Gregory_Bonito)
* [Gian Maria Niccol√≤ Benucci](https://github.com/Gian77)

<a name="abstract"></a>
Abstract
--------
One of the most crucial steps in high-throughput sequence based microbiome studies is the taxonomic assignment of sequences belonging to operational taxonomic units (OTUs). Without taxonomic classification, functional and biological information of microbial communities cannot be inferred or interpreted. The Internal Transcribed Spacer (ITS) region of the nuclear ribosomal DNA (rDNA) is the conventional marker region for fungal community studies. While bioinformatics pipelines that cluster reads into OTUs have received much attention in the literature, the taxonomic classification of these sequences, upon which biological inference is dependent, has been largely neglected. Here we compared how the most common fungal OTU taxonomic assignment approaches, UTAX, RDP standalone, and SINTAX, handle ITS1 and ITS2 fungal sequence data generated by Illumina MiSeq amplicon sequencing. The classification power (ability to assign taxonomy at a given taxonomic rank) varied among the classifiers. Our results demonstrate how taxonomy assignment tools can bias understanding of fungal communities, and distort the linkage between taxa and specific treatments or ecosystems. We conclude with general guidelines and best practices for assigning taxonomy in fungal microbiome studies.

<a name="study-purpose"></a>
Study Purpose
--------
We set out to test whether the most commonly used taxonomic classifiers generate similar profiles of the fungal community. To do so we used two published MiSeq ITS datasets and compared the standalone command line [Ribosomal Database Project Classifier](http://rdp.cme.msu.edu/) (Wang et al. 2007; Cole et al. 2013),  [UTAX](http://www.drive5.com/usearch/manual/utax_algo.html) (Edgar 2010; 2013), and [SINTAX](http://biorxiv.org/content/early/2016/09/09/074161) (Edgar 2016).

<a name="pipeline"></a>
Pipeline
--------
The analysis requires four general steps: 
(1) Sequence QC and OTU-picking 
(2) Pre-taxonomy assignment database formatting and training
(3) Taxonomy assignment
(4) Post-taxonmy assignment OTU processing and filtering

<a name="script-purpose"></a>
Script Purpose
--------
otu_processing.sh - steps used in this study to sub-sample reads, filter and quality-control reads, assign reads to OTUs, and sub-sample OTUs. 
MasterScript1.py - The first Master script in this pipeline accepts database files in FASTA format and re-generates that database in formats compatible with the RDP, UTAX, and SINTAX classifiers. It also calls on two scripts to generate files for training the RDP classifier.
Taxonomy_pipeline.sh - The second script requires previous installation of RDP, UTAX, and SINTAX. Taxonomy inputs and outputs will be in the same folder by default. 
MasterScript2.py - The second Master script accepts the classified OTU taxonomy outputs from each of the three classifiers, filters by confidence score, and reformats them to a standardized output. In addition, it compares the standardized classifications to generate a consensus taxonomy based on all three classifiers.

<a name="sys-req"></a>
System Requirements
--------
* Python version 2.6
* [PEAR version 0.9.8](http://sco.h-its.org/exelixis/web/software/pear/) (Zhang et al. 2014)
* [USEARCH version 8](http://drive5.com/usearch/manual8.1/) (Edgar et al. 2011; Edgar 2013)
* [RDP version 11](https://github.com/rdpstaff/classifier) (Cole et al. 2013)
* [USEARCH version 9](http://drive5.com/usearch/manual/whatsnewv9.html) (Edgar 2016)
