---
title: "Fungal microbiomes: a strategy for improved taxonomic resolution of environmental ITS sequences"
author: "Supplemental File 1"
date: "Gdanetz, VandePol, Bonito, and Benucci"
output: html_document
---

## The analysis requires four general steps: 
1. Sequence QC and OTU-picking 
2. Database formatting and training 
3. Taxonomy assignment 
4. Post-taxonomy-assignment processing, filtering, analyzing 

***
## __Step 1:__ Sequence QC and OTU-Picking.
If beginning with raw sequence files, follow the OTU-picking steps below. If beginning with OTUs, skip to Step 2.

#### Required Programs:
* Python version 2.6 (or higher)
* PEAR version 0.9.8 
* USEARCH version 8 
* RDP version 11 
* USEARCH version 9 
* R and RStudio (newest version)

#### Run the Code:
```{bash eval=FALSE}
## get quality statistics
fastqc ./*_R1_001.fastq
fastqc ./*_R2_001.fastq

## merge forward and reverse, change labels
usearch8 -fastq_mergepairs *_R1.fq -relabel @ -tabbedout merged_tabbed.txt -report merged_summary.txt -fastqout merged.fastq

## subsample reads
#download from https://github.com/lh3/seqtk
seqtk sample -s100 merged.fastq 500000 > sub.merged.fastq

## quality filter
usearch8 -fastq_eestats2 sub.merged.fastq -output merged.pre_filtered.eestats2.txt -length_cutoffs 100,400,10
usearch8 -fastq_filter sub.merged.fastq -fastq_minlen 150 -fastq_maxee 0.5 -fastaout merged.filtered.fa -fastaout_discarded merged.no_filter.fa -fastqout merged.filtered.fastq
usearch8 -fastq_eestats2 merged.filtered.fastq -output merged.post_filtered.eestats2.txt -length_cutoffs 150,400,10

## trim cleaned reads
usearch8 -fastx_truncate merged.filtered.fastq -padlen 250 -trunclen 250 -fastaout cleaned.fa

## find unique/representative sequences
usearch8 -derep_fulllength cleaned.fa -sizeout -fastaout derep.fa -threads 4

## cluster OTUs, remove singletons, de novo chimera check
usearch8 -cluster_otus derep.fa -minsize 2 -sizein -sizeout -relabel OTU_ -otus derep.otus.fa -uparseout reads.derep.otus.txt

## reference-based chimera check
# use for ITS1 region: uchime_sh_refs_dynamic_develop_985_01.01.2016.ITS1.fasta
# use for ITS2 region: uchime_sh_refs_dynamic_develop_985_01.01.2016.ITS2.fasta
# wget https://unite.ut.ee/sh_files/uchime_reference_dataset_01.01.2016.zip
usearch8 -uchime_ref derep.otus.fa -db /path_to_files/UNITE_db/uchime_sh_refs_dynamic_develop_985_01.01.2016.ITS2.fasta -nonchimeras otus.no_chimera.fa -uchimeout reads.derep.otus.no_chimera.uchime -strand plus -sizein -sizeout

## map reads back to OTUs and create the otu_tab.txt for phyloseq
usearch8 -usearch_global merged.filtered.fa -db otus.no_chimera.fa -strand plus -id 0.97 -top_hit_only -otutabout otu_table.txt -sizein -sizeout

## subsample OTUs to 500
seqtk sample -s100 otus.no_chimera.fa 500 > OTU_500.fa
```
#### Output Files:
* fasta file with all OTUs (otus.no_chimera.fa)
* subsetted fasta file (OTU_500.fa)

***
## __Step 2:__ Pre-taxonomy-assignment database formatting and training.
#### Download the following python scripts from GitHub or supplemental files:
* _subscript_lineage2taxonomyTrain.py_
* _subscript_fasta_addFullLineage.py_ 
* _FormatRefDB.py_

#### Obtain a copy of the UNITE fungal database:
[Download database here](https://unite.ut.ee/repository.php) 

#### Install the RDP Classifier following instructions available here:
[RDP Install](https://github.com/rdpstaff) 

#### Run the Scripts.
Three python scripts listed above must be in the same directory with the database file:
```{python eval=FALSE}
python FormatRefDB.py
```
#### Output Files:
1. Output files from utax database formatting 
* utax_train.log
* utax_report.txt
* make_udb.log
* utax.31.01.2016.tc
* utax_db_31.01.2016.db
2. Output files from RDP formatting
* mytrained
* train.fasta
* train.txt 
3. Output files from SINTAX formatting:
* none 

***
## __Step 3:__ Taxonomy assignment.
#### Input Files:
* Database files generated in Step 2
* OTU table(s)

#### Run Commands: 
```{bash eval=FALSE}
## assign taxonmy with utax
usearch8 -utax /path_to_files/RDP-UTAX/OTU_500.fa -db /path_to_files/unite_ref_db/utax_db_31_01_2016.db -strand both -utaxout /path_to_files/OTU_500.utax -utax_cutoff 0.8 -threads 6

## assign taxonomy with sintax 
usearch9 -sintax /path_to_files/OTU_500.fa -db sintax_31_01_2016.udb -tabbedout /path_to_files/OTU_500.sintax -strand both -sintax_cutoff 0.8

## assign taxonmy with RDP Classifier
cd /path_to_files/RDP_installation/

java -Xmx5g -jar /path_to_files/mytrained/classifier.jar classify --conf 0.8 --format allrank --train_propfile /path_to_files/mytrained/rRNAClassifier.properties -o /path_to_files/OTU_500.rdp /path_to_files/OTU_500.fa
```
#### Output Files: Three taxonomy tables.
1. OTU_500.rdp
2. OTU_500.utax
3. OTU_500.sintax

***
## __Step 4:__ Post-taxonomy-assignment OTU processing, filtering, analyzing.

#### Download the following python and R scripts from GitHub or supplemental files.
* _CombineTaxonomy.py_
* _Import.R_ 

### Run _CombineTaxonomy.py_
```{python eval=FALSE}
python CombineTaxonomy.py 
```

This script is interactive. First users will be prompted for cutoff threshold to filter taxonomy tables (the default is 0.8):
![](/Users/Home/Documents/Google_Drive/RDP_vs_Utax_Project/Drafts_of_Figures/screen_shots/prompt1.png) 


Then users will be prompted for the locations of each of the classifier outputs:
![](/Users/Home/Documents/Google_Drive/RDP_vs_Utax_Project/Drafts_of_Figures/screen_shots/prompt2.png) 


Finally the script will ask if the user would like to modify the default file names for the outputs: 
![](/Users/Home/Documents/Google_Drive/RDP_vs_Utax_Project/Drafts_of_Figures/screen_shots/prompt3.png) 


#### Format taxonomy outputs.
The taxonomy files from each classifier will modified to a new format, and all column labels will be identical:

![unfiltered RDP taxonomy table](/Users/Home/Documents/Google_Drive/RDP_vs_Utax_Project/Drafts_of_Figures/screen_shots/rdp.png) 

![final RDP taxonomy table](/Users/Home/Documents/Google_Drive/RDP_vs_Utax_Project/Drafts_of_Figures/screen_shots/rdp-final.png) 



#### Build consensus taxonomy.
The _CombineTaxonomy.py_ script generates a *combined_taxonomy.txt* file that places all taxonomy table columns side-by-side in one larger taxonomy table: 
![](/Users/Home/Documents/Google_Drive/RDP_vs_Utax_Project/Drafts_of_Figures/screen_shots/combined.png) 



_CombineTaxonomy.py_ also generates a *consensus_taxonomy.txt* file which parses over the *combined_taxonomy.txt* file, and selects the improved taxanomic classification using the rules outlined in the manuscript text (Table 3):
![](/Users/Home/Documents/Google_Drive/RDP_vs_Utax_Project/Drafts_of_Figures/screen_shots/consensus.png) 

### Downstream Analysis.

#### Import files into R: 
Run _Import.R_ which imports data tables. Requires _combined_taxonomy.txt_ tables. 

First, load R packages:
```{r eval=FALSE}
library(dplyr)
library(ggplot2)
```

Import file(s) and format data frame(s):
```{r eval=FALSE}
data = read.table("~/path_to_files/combined_taxonomy.txt", header=TRUE, row.names=1, sep="\t")

data[data==''|data==' ']<-NA

sapply(data, function(x) sum(is.na(x))) -> unassigned_data
```

Calculate unidentified OTUs, assigned OTUs, and further format the data frame(s):
```{r eval=FALSE}
data2 <- as.data.frame(unassigned_data)

data2$Classifier <- c("RDP","SINTAX","UTAX","COMBINED","RDP","SINTAX","UTAX","COMBINED","RDP","SINTAX","UTAX","COMBINED","RDP","SINTAX","UTAX","COMBINED","RDP","SINTAX","UTAX","COMBINED","RDP","SINTAX","UTAX","COMBINED","RDP","SINTAX","UTAX","COMBINED")

data2$Rank <- row.names(data2)

data2$Assigned <- sqrt((data2$unassigned_data -500)^2) #change 500 to number of OTUs

data2$Classifier <- factor(data2$Classifier, levels = c("RDP","UTAX","SINTAX","COMBINED"))
```

#### Generate Barplots: 
Use ggplot to generate barplots as presented in Figure 2. 
```{r eval=FALSE}
ggplot(data2, aes(x = Rank, y = Assigned, fill= Classifier)) + 
  geom_bar(stat = "identity") +
  scale_x_discrete(limits=data2$Rank) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1),
        panel.grid=element_blank(),
        panel.background=element_blank()) +
  ggtitle("Assigned and Unidentified OTUs in ITS Dataset") +
  labs(x="Taxonomic Ranks", y="Number of classified OTUs")
```

Repeat R code above for each ITS dataset.

***